core/templates/pages/about-page/about-page.component.ts

```diff
@@ -19,100 +19,77 @@
-    private windowRef: WindowRef) {
+    private windowRef: WindowRef,
+    private siteAnalyticsService: SiteAnalyticsService,
+    private userService: UserService,
+    private loaderService: LoaderService) {
   }
 
-  getCredits(startLetter: string): string[] {
-    const results = AboutPageConstants.CREDITS_CONSTANTS.filter(
-      (credit) => credit.startsWith(startLetter)).sort();
-    return results;
+  getStaticImageUrl(imagePath: string): string {
+    return this.urlInterpolationService.getStaticImageUrl(imagePath);
   }
 
-  onTabClick(tabName: string): Window {
-    this.windowRef.nativeWindow.location.hash = '#' + tabName;
-    this.activeTabName = tabName;
-    return this.windowRef.nativeWindow;
+  onClickVisitClassroomButton(): void {
+    this.siteAnalyticsService.registerClickVisitClassroomButtonEvent();
+    this.windowRef.nativeWindow.location.href = this.classroomUrl;
   }
 
-  getStaticImageUrl(imagePath: string): string {
-    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  onClickBrowseLibraryButton(): void {
+    this.siteAnalyticsService.registerClickBrowseLibraryButtonEvent();
+    this.windowRef.nativeWindow.location.href = '/community-library';
   }
 
-  ngOnInit(): void {
-    this.activeTabName = this.TAB_ID_ABOUT;
-    this.allCredits = [];
-    const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
-    for (const letter of letters) {
-      const names = this.getCredits(letter);
-      if (names.length > 0) {
-        this.allCredits.push({letter, names});
-      }
-    }
-    const hash = this.windowRef.nativeWindow.location.hash.slice(1);
-    if (hash === 'license') {
-      this.activeTabName = this.TAB_ID_FOUNDATION;
-    } else if (this.ALLOWED_TABS.includes(hash)) {
-      this.activeTabName = hash;
-    }
-
-    this.listOfNames = this.listOfNamesToThank
-      .slice(0, this.listOfNamesToThank.length - 1).join(', ') +
-      ' & ' + this.listOfNamesToThank[this.listOfNamesToThank.length - 1];
-    this.aboutPageMascotImgUrl = this.urlInterpolationService
-      .getStaticImageUrl('/general/about_page_mascot.webp');
+  onClickCreateLessonButton(): void {
+    this.siteAnalyticsService.registerCreateLessonButtonEvent();
+    this.windowRef.nativeWindow.location.href = (
+      '/creator-dashboard?mode=create');
+  }
 
-    this.windowRef.nativeWindow.onhashchange = () => {
-      const hashChange = this.windowRef.nativeWindow.location.hash.slice(1);
-      if (hashChange === 'license') {
-        this.activeTabName = this.TAB_ID_FOUNDATION;
-        this.windowRef.nativeWindow.location.reload(true);
-      } else if (this.ALLOWED_TABS.includes(hashChange)) {
-        this.activeTabName = hashChange;
-      }
-    };
+  ngOnInit(): void {
+    this.classroomUrl = this.urlInterpolationService.interpolateUrl(
+      '/learn/<classroomUrlFragment>', {
+        classroomUrlFragment: constants.DEFAULT_CLASSROOM_URL_FRAGMENT
+      });
```

Comment:
1. Why do we need to tie classroomUrl to this? is it used in the component's template?
2. Why is this need to be initialized here? I only see this to be used inside onClickVisitClassroomButton, can we define it the function scope?

Reply:
shifted in onClickVisitClassroomButton() function

Reply:
Do we still need `this.classroomUrl` or a simple `let classroomUrl` variable would work?

Reply:
changed to let.